arrow/parser.py

```diff
@@ -95,23 +106,14 @@ def __init__(self, locale="en_us", cache_size=0):
     # TODO: since we support more than ISO-8601, we should rename this function
     # IDEA: break into multiple functions
     def parse_iso(self, datetime_string):
-        # strip leading and trailing whitespace
-        datetime_string = datetime_string.strip()
-
+        # TODO: add a flag to normalize whitespace (useful in logs, ref issue #421)
         has_space_divider = " " in datetime_string
         has_t_divider = "T" in datetime_string
 
         num_spaces = datetime_string.count(" ")
-        if has_space_divider and num_spaces != 1:
+        if has_space_divider and num_spaces != 1 or has_t_divider and num_spaces > 0:
```

Comment:
Agreed, and I liked that at first! But after some thought, I was thinking that it may be better to keep the message broad ("Expected an ISO 8601-like string, but was given '____'. Try passing in a format string to resolve this.").

I'd be down for reverting this change, but I thought it would be worth a discussion.